kernel/mempool.c

```diff
@@ -33,7 +33,7 @@ static void *z_heap_aligned_alloc(struct k_heap *heap, size_t align, size_t size
 
 	heap_ref = mem;
 	*heap_ref = heap;
-	mem = ++heap_ref;
+	mem = heap_ref + 1;
```

Comment:
`heap_ref` still needs to be incremented.

Reply:
I don't see any use of local variable heap_ref after this point... Am I missing something?